iib/workers/tasks/utils.py

```diff
@@ -175,28 +176,49 @@ def _requires_max_ocp_version(bundle):
     return False
 
 
-def serve_index_registry(db_path):
+def serve_registry(base_dir, from_index=None, fbc_dir=None, db_path=None):
     """
     Locally start OPM registry service, which can be communicated with using gRPC queries.
 
     Due to IIB's paralellism, the service can run multiple times, which could lead to port
     binding conflicts. Resolution of port conflicts is handled in this function as well.
 
+    :param str base_dir: base directory to create temporary files in.
+    :param str from_index: index image to inspect.
+    :param str fbc_dir: path to file-based catalog directory.
     :param str db_path: path to index database containing the registry data.
     :return: tuple containing port number of the running service and the running Popen object.
     :rtype: (int, Popen)
     :raises IIBError: if all tried ports are in use, or the command failed for another reason.
     """
+    from iib.workers.tasks.build import _get_index_database
+
     conf = get_worker_config()
     port_start = conf['iib_grpc_start_port']
     port_end = port_start + conf['iib_grpc_max_port_tries']
 
+    assert (
+        len([i for i in (from_index, fbc_dir, db_path) if i]) == 1
+    ), "Only one keyword parameter can be set"
+
+    if from_index:
+        if not is_image_fbc(from_index):
```

Comment:
If we use get_image_label we don't have to use this function

Reply:
That is true. I did not thought about it at first. On the other hand, this will prevent having the long name of the label in multiple places. But I guess you are right, we could have done that.

Reply:
any update here?

Reply:
I think we had a PR about this one, but it was closed later

Reply:
Because  I am torn between two solutions ... should we have `is_image_fbc` or should we just have the FBC specific LABEL as a constant and use  `get_image_label` directly (on all places where `is_image_fbc` is)

Reply:
I believe using `is_image_fbc` keeps the code more readable and decoupled. 

It's better to keep the "intelligence" in `is_image_fbc` than using a constant IMO since today it's a simple label, one day it can be something more complex than that and the effort to change a lot of functions will be greater than simply changing `is_image_fbc` behavior.

Reply:
yeah, lets agree on keeping `is_image_fbc` for now. however I will modify it to use `get_image_label` inside. In different PR.

Reply:
I know I bother you so much with this but.. Could you create a task or PR and share it here? I worry this will get lost. Feel free to close this conversation afterward.

Reply:
https://github.com/release-engineering/iib/pull/328